lib/compress/zstd_double_fast.c

```diff
@@ -175,9 +175,13 @@ size_t ZSTD_compressBlock_doubleFast_noDict_generic(
         } while (ip1 <= ilimit);
 
 _cleanup:
+        /* If offset_1 started invalid (offsetSaved1 > 0) and became valid (offset_1 > 0),
+         * rotate saved offsets. */
```

Comment:
Please add a more complete explaination. You can just add it to one match finder, e.g. fast, and then add a brief summary in the other match finders, and point to the full explaination.

Maybe something like this, but feel free to re-word as you see fit.

```suggestion

        /* When the repcodes are outside of the prefix, we set them to zero before the loop.
         * When the offsets are still zero, we need to restore them after the block to have a correct
         * repcode history. If only one offset was invalid, it is easy. The tricky case is when both
         * offsets were invalid. We need to figure out which offset to refill with. If both offsets
         * are zero they are in the same order. If both offsets are non-zero, we won't restore the
         * offsets from `offsetSaved[12]`. If only one is zero, we need to decide which offset to
         * restore. If offset_1 is non-zero, then offset_2 must be offsetSaved1. It is impossible for
         * offset_2 to be non-zero.
         * So if offset_1 started invalid (offsetSaved1 > 0) and became valid (offset_1 > 0), then
         * set rep[0] = offset_1 and rep[1] = offsetSaved1.
         */
```
```suggestion
        /* If offset_1 started invalid (offsetSaved1 > 0) and became valid (offset_1 > 0),
         * rotate saved offsets. */
```